firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift

```diff
@@ -0,0 +1,902 @@
+
+        lockImageHeightConstraint?.constant = min(UIFontMetrics.default.scaledValue(for: iconSize), 2 * iconSize)
+        connectionArrowHeightConstraint?.constant = min(UIFontMetrics.default.scaledValue(for: iconSize), 2 * iconSize)
+
+        view.setNeedsLayout()
+        view.layoutIfNeeded()
+    }
+
+    // MARK: - Button actions
+    @objc
+    func closeButtonTapped() {
+        enhancedTrackingProtectionMenuDelegate?.didFinish()
+    }
+
+    @objc
+    func connectionDetailsTapped() {
+//        let detailsVC = TrackingProtectionDetailsViewController(with: viewModel.getDetailsViewModel(),
+//                                                                windowUUID: windowUUID)
+//        detailsVC.modalPresentationStyle = .pageSheet
+//        self.present(detailsVC, animated: true)
+    }
+
+    @objc
+    func blockedTrackersTapped() {}
+
+    @objc
+    func trackingProtectionToggleTapped() {
+        // site is safelisted if site ETP is disabled
+        viewModel.toggleSiteSafelistStatus()
+        toggleStatusLabel.text = toggleSwitch.isOn ? .ETPOn : .ETPOff
+    }
+
+    @objc
+    private func didTapClearCookiesAndSiteData() {
+        viewModel.onTapClearCookiesAndSiteData(controller: self)
+    }
+
+    @objc
+    func protectionSettingsTapped() {
+        enhancedTrackingProtectionMenuDelegate?.settingsOpenPage(settings: .contentBlocker)
+    }
+
+    // MARK: - Gesture Recognizer
+    private func addGestureRecognizer() {
+        let panGesture = UIPanGestureRecognizer(target: self, action: #selector(panGestureRecognizerAction))
+        view.addGestureRecognizer(panGesture)
+    }
+
+    @objc
+    func panGestureRecognizerAction(sender: UIPanGestureRecognizer) {
+        let translation = sender.translation(in: view)
+        let originalYPosition = self.view.frame.origin.y
+        let originalXPosition = self.view.frame.origin.x
+
+        // Not allowing the user to drag the view upward
+        guard translation.y >= 0 else { return }
+
+        // Setting x based on window calculation because we don't want
+        // users to move the frame side ways, only straight up or down
+        view.frame.origin = CGPoint(x: originalXPosition,
+                                    y: self.pointOrigin!.y + translation.y)
+
+        if sender.state == .ended {
+            let dragVelocity = sender.velocity(in: view)
+            if dragVelocity.y >= 1300 {
+                enhancedTrackingProtectionMenuDelegate?.didFinish()
+            } else {
+                // Set back to original position of the view controller
+                UIView.animate(withDuration: 0.3) {
+                    self.view.frame.origin = self.pointOrigin ?? CGPoint(x: originalXPosition, y: originalYPosition)
+                }
+            }
+        }
+    }
+
+    private func currentTheme() -> Theme {
+        return themeManager.getCurrentTheme(for: windowUUID)
+    }
+
+    func willDismiss() {
+    }
```

Comment:
should we add a TODO with a ticket here?

Reply:
Nope, deleted.

Reply:
No, we don't.